modules/template/template.go

```diff
@@ -0,0 +1,40 @@
+package template
+
+import (
+	"context"
+	"strings"
+
+	"github.com/risor-io/risor/object"
+)
+
+func Template(ctx context.Context, args ...object.Object) object.Object {
+	numArgs := len(args)
+
+	if numArgs != 2 {
+		return object.NewArgsError("template", 2, numArgs)
+	}
+
+	data, argsErr := object.AsMap(args[0])
+	if argsErr != nil {
+		return argsErr
+	}
+
+	template, argsErr := object.AsString(args[1])
+	if argsErr != nil {
+		return argsErr
+	}
+
+	buf := new(strings.Builder)
+
+	if err := Render(buf, template, data.Interface()); err != nil {
+		return object.NewError(err)
+	}
+
+	return object.NewString(buf.String())
+}
+
+func Builtins() map[string]object.Object {
+	return map[string]object.Object{
+		"template": object.NewBuiltin("template", Template),
```

Comment:
The `template` function seems to work great.

As someone that just gave it a try without much review first, at first I didn't expect to have to use `.Values` as the prefix to use the input data. To what degree do you think of this @luisdavim as a _Helm_ template equivalent vs. a Go template equivalent?

I wonder if both should be offered in Risor. If both were offered, would it make sense for the Helm version to be known as the global builtin `template` or would that most logically correspond to Go's `text/template` package?

I'm not sure of the "right" answer yet... this is just something I'm thinking through real quick.

Reply:
The main reason ti place the passed values under `Values` (or any other field) is to keep them separate from any builtin sets of values we might have and avoid collisions, in this case we have the `Env` builtin set of values and in the future we might want to add more...

I'm ok with renaming or even removing both the `.Values` and `.Env`....

Reply:
I think having one builtin that operates exactly in this way makes a ton of sense and accessing `.Env` values will be super useful in Risor, I expect.

But we may want a separate builtin that is a vanilla Go text/template without .Values and .Env. 

So if we have both options, I'm not yet sure what the best names for each would be. And whether one or both should be offered in a `template` module. If we did a module we could still have the module object itself be callable, which is something I did in one or two modules already.

Reply:
Yeah but if we do have this option with "batteries included" (out of the box values and functions) is it really need to also provide a more barebones option that is closer to what Go text/template provides? Do you have any use cases for that in mind?

Reply:
The `.Values` approach only has the downsides that 1) it's not intuitively guessable and 2) it's slightly more typing when you're just feeding in a simple map.

If someone comes to Risor expecting the `template` builtin to correspond to Go's `text/template` package, which would be reasonable in general, they would have no idea that they have to use `.Values` in their template.

I'm totally fine with having this version of the `template` builtin to be the one and only version for now. But I'm just wanting to allow for the other version to be added later without creating naming confusion.

Reply:
OK, what about some approach like what we have with the http module, a built-in that is the current implementation, maybe named render_template or just render and a module that allows initiate a template, set the delimiters and maybe some other options and then render a template, this one would be closer to the default golang template and have no default values.

BTW, I agree the .Values is not intuitive or easily discoverable nut that also a matter of documentation.

Reply:
Yeah, just using the name “render” - for what you have here - would be enough for now. That’ll let us reserve the name “template” for the module, whenever we do that.

If you just do that simple rename we can get this PR merged.

Having this builtin named as a verb makes some sense too, another plus.

Reply:
@myzie I've updated the PR, renamed the original builtin to `render` and added a `template` module, we can add things like `parse_files` to it latter but I think this is a good starting point, please let me know what you think.